src/test/kotlin/org/jetbrains/intellij/tasks/DownloadIntelliJSpec.kt

```diff
@@ -5,100 +5,109 @@ import org.jetbrains.intellij.IntelliJPluginSpecBase
 import java.io.File
 import kotlin.test.Test
 import kotlin.test.assertEquals
+import kotlin.test.assertFalse
 import kotlin.test.assertTrue
 
 class DownloadIntelliJSpec : IntelliJPluginSpecBase() {
```

Comment:
If I understood these tests correctly, they were fragile because files could exist because other tests, executed before, downloaded them. I'm making sure all relevant stuff is deleted in the setup method. Tests are slower, but safe now.

Also:
- refactored tests for better readability and added one case so we have now downloading ultimate IDE with and without sources
- changed IU 14.1.1 to 2021.2.4 to save some disk space on the runners

Reply:
This is updated now to use unique IDEs versions. Unfortunately, deleting directories before tests fail on Windows runners, probably due to the daemon locking the files, which shouldn't be run at all in our setup. It may be caused by Gradle issues, see:
- https://github.com/gradle/gradle/issues/937
- https://github.com/gradle/gradle/issues/12535

I'm leaving this approach for now, which is better than nothing, and will monitor the above issues. We can improve it in the future.